modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -0,0 +1,463 @@
+        refs: 'CVE-2015-4000',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # Null ciphers
+    if ssl_cipher.upcase.include? 'NULL'
+      print_good('Accepts Null cipher')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed Null cipher.",
+        refs: 'CVE-2011-3389',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # deprecation
+    if ssl_version == 'SSLv2'
+      print_good('Accepts Deprecated SSLv2')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv2. Which was deprecated in 2011",
+        refs: 'https://datatracker.ietf.org/doc/html/rfc6176',
+        exploited_at: Time.now.utc
+      )
+    elsif ssl_version == 'SSLv3'
+      print_good('Accepts Deprecated SSLv3')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv3. Which was deprecated in 2015",
+        refs: 'https://datatracker.ietf.org/doc/html/rfc7568',
+        exploited_at: Time.now.utc
+      )
+    elsif ssl_version == 'TLSv1.0'
+      print_good('Accepts Deprecated TLSv1.0')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed TLSv1.0. Which was widely deprecated in 2020",
+        refs: 'https://datatracker.ietf.org/doc/html/rfc7568',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    return if cert.nil?
+
+    key_size = public_key_size(cert)
+    if key_size > 0
+      if key_size == 1024
+        print_good('Public Key only 1024 bits')
+        report_vuln(
+          host: ip,
+          port: rport,
+          proto: 'tcp',
+          name: name,
+          info: "Module #{fullname} confirmed certificate key size 1024 bits",
+          refs: references,
+          exploited_at: Time.now.utc
+        )
+      elsif key_size < 1024
+        print_good('Public Key < 1024 bits')
+        report_vuln(
+          host: ip,
+          port: rport,
+          proto: 'tcp',
+          name: name,
+          info: "Module #{fullname} confirmed certificate key size < 1024 bits",
+          refs: references,
+          exploited_at: Time.now.utc
+        )
+      end
+    end
+
+    # certificate signed md5
+    alg = cert.signature_algorithm
+
+    if alg.downcase.include? 'md5'
+      print_good('Certificate signed with MD5')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed certificate signed with MD5 algo",
+        refs: references,
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # expired
+    if cert.not_after <= DateTime.now
```

Comment:
Would this not want to be `<` since technically if they are the same its still valid for that brief period of time?